services/src/main/java/org/keycloak/userprofile/validator/MultiValueValidator.java

```diff
@@ -0,0 +1,139 @@
+import org.keycloak.validate.validators.ValidatorConfigValidator;
+
+public class MultiValueValidator implements SimpleValidator, ConfiguredProvider {
+
+    public static final String ID = "multivalued";
+    public static final String MESSAGE_INVALID_SIZE = "error-invalid-multivalued-size";
+    public static final String KEY_MIN = "min";
+    public static final String KEY_MAX = "max";
+
+    @Override
+    public String getId() {
+        return ID;
+    }
+
+    @Override
+    public ValidationContext validate(Object value, String inputHint, ValidationContext context, ValidatorConfig config) {
+        AttributeContext attributeContext = UserProfileAttributeValidationContext.from(context).getAttributeContext();
+
+        Long min = Optional.ofNullable(config.getLong(KEY_MIN)).orElse(getDefaultMinSize(context));
+        Long max = config.getLong(KEY_MAX);
+
+        if (!(value instanceof Collection)) {
+            addError(inputHint, context, min, max);
+            return context;
+        }
+
+        long length = ((Collection<String>) value).stream().filter(StringUtil::isNotBlank).count();
+
+        if (length == 0 && attributeContext.getMetadata().isRequired(attributeContext)) {
+            // if no value is set and attribute is required, do not validate in favor of the required validator
+            return context;
+        }
+
+        if (!(length >= min && length <= max)) {
+            addError(inputHint, context, min, max);
+            return context;
+        }
+
+        return context;
+    }
+
+    @Override
+    public ValidationResult validateConfig(KeycloakSession session, ValidatorConfig config) {
+        if (ValidatorConfig.isEmpty(config)) {
+            return ValidationResult.of(new ValidationError(ID, KEY_MAX, MESSAGE_CONFIG_MISSING_VALUE));
+        }
+
+        Set<ValidationError> errors = new HashSet<>();
+        Long min = config.getLong(KEY_MIN);
+        Long max = config.getLong(KEY_MAX);
+
+        if (min == null && config.containsKey(KEY_MIN)) {
+            errors.add(new ValidationError(ID, KEY_MIN, MESSAGE_CONFIG_INVALID_NUMBER_VALUE));
+        }
+
+        if (max == null) {
+            errors.add(new ValidationError(ID, KEY_MAX,
+                    config.containsKey(KEY_MAX) ? MESSAGE_CONFIG_INVALID_NUMBER_VALUE : MESSAGE_CONFIG_MISSING_VALUE));
+        } else if (min != null && (min > max)) {
+            errors.add(new ValidationError(ID, KEY_MAX, ValidatorConfigValidator.MESSAGE_CONFIG_INVALID_VALUE));
+        }
+
+        return ValidationResult.of(errors);
+    }
+
+    @Override
+    public String getHelpText() {
+        return "Multivalued validator";
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigProperties() {
+        List<ProviderConfigProperty> properties = new ArrayList<>();
+        ProviderConfigProperty property;
+        property = new ProviderConfigProperty();
+        property.setName(KEY_MIN);
+        property.setLabel("Minimum size");
```

Comment:
Nitpick: Is it better to use `Minimum count` instead of `Minimum size` ? Same for "max" and corresponding help texts.

Reply:
Not sure. I would leave as is.

Reply:
Sounds good